core/src/main/scala/org/scalatra/servlet/RichRequest.scala

```diff
@@ -112,31 +112,23 @@ case class RichRequest(r: HttpServletRequest) extends AttributesMap {
    * A map of headers.  Multiple header values are separated by a ','
    * character.  The keys of this map are case-insensitive.
    */
-  object headers extends collection.immutable.Map[String, String] {
+  object headers {
 
     def get(name: String): Option[String] = Option(r.getHeader(name))
 
-    override def +[V1 >: String](kv: (String, V1)): Map[String, V1] = {
-      val b = Map.newBuilder[String, V1]
-      b ++= this
-      b += ((kv._1, kv._2))
-      b.result()
+    def getOrElse(name: String, default: => String): String = {
+      headers.get(name) match {
+        case Some(v) => v
+        case None => default
+      }
     }
 
-    override def -(key: String): Map[String, String] = {
-      val b = this.newBuilder
-      for (kv <- this; if kv._1 != key) b += kv
-      b.result()
-    }
+    def keys: Iterator[String] = r.getHeaderNames.asScala
```

Comment:
`names` might be better?

Reply:
That is more appropriate.
I change the method name to `names`.